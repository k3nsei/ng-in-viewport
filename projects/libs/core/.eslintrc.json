{
  "extends": "../../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "ngx-intersection",
            "style": "kebab-case"
          }
        ],
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "ngxIntersection",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/no-host-metadata-property": "off"
      }
    },
    {
      "files": ["*.html"],
      "extends": ["plugin:@angular-eslint/template/recommended", "plugin:@angular-eslint/template/accessibility"],
      "rules": {}
    }
  ]
}
