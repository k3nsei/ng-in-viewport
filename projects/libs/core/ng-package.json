{
  "$schema": "../../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../../dist/libs/core",
  "lib": {
    "entryFile": "src/public-api.ts"
  },
  "allowedNonPeerDependencies": ["@ngx-intersection/utils"]
}
