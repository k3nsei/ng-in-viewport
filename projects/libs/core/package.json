{
  "name": "@ngx-intersection/core",
  "version": "0.1.0",
  "sideEffects": false,
  "dependencies": {
    "@ngx-intersection/utils": "^0.1.0",
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^18.0.0",
    "@angular/core": "^18.0.0"
  }
}
