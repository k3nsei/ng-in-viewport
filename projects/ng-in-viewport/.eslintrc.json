{
  "extends": "../../.eslintrc.json",
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "rules": {
        "@typescript-eslint/no-explicit-any": "off",
        "@typescript-eslint/no-non-null-assertion": "off",
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "inViewport",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "in-viewport",
            "style": "kebab-case"
          }
        ]
      }
    },
    {
      "files": ["*.html"],
      "rules": {}
    }
  ]
}
