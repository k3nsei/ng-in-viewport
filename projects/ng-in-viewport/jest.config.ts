import type { Config } from 'jest';

const jestConfig: Config = {
  displayName: 'ng-in-viewport',
  preset: 'jest-preset-angular',
  // globalSetup: 'jest-preset-angular/global-setup',
  setupFilesAfterEnv: ['<rootDir>/jest-setup.ts'],
  coverageDirectory: '<rootDir>/../../coverage/ng-in-viewport',
  coverageProvider: 'v8',
  coverageReporters: ['lcovonly', 'text', 'html-spa'],
  coverageThreshold: {
    global: {
      branches: 95,
      functions: 95,
      lines: 90,
      statements: 90,
    },
  },
  coveragePathIgnorePatterns: ['node_modules/', 'enums/', 'index.ts', 'public-api.ts', 'in-viewport.module.ts'],
};

export default jestConfig;
