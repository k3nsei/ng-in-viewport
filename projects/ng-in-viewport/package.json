{
  "name": "ng-in-viewport",
  "version": "15.0.2",
  "description": "Allows us to check if an element is within the browsers visual viewport",
  "homepage": "https://k3nsei.github.io/ng-in-viewport/",
  "bugs": {
    "url": "https://github.com/k3nsei/ng-in-viewport/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/k3nsei/ng-in-viewport.git"
  },
  "license": "MIT",
  "author": {
    "name": "Piotr StÄ™pniewski",
    "email": "k3nsei.pl@gmail.com",
    "url": "https://github.com/k3nsei"
  },
  "dependencies": {
    "lodash-es": "^4.17.21",
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^14.0.0 || >=15.0.0",
    "@angular/core": "^14.0.0 || >=15.0.0"
  },
  "publishConfig": {
    "provenance": true
  }
}
