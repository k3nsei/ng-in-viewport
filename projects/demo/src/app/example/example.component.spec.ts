import { DebugElement, NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { InViewportAction } from 'ng-in-viewport';

import { ExampleComponent } from './example.component';

describe('ExampleComponent', () => {
  const activeClassName = 'item--active';
  let component: ExampleComponent;
  let fixture: ComponentFixture<ExampleComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ExampleComponent],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();

    fixture = TestBed.createComponent(ExampleComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create component', () => {
    const actual: boolean = component instanceof ExampleComponent;
    const expected = true;

    expect(actual).toBe(expected);
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it(`should have correct elements count`, () => {
    const actual: number = component.elements.length;
    const expected = 100;

    expect(actual).toBe(expected);
  });

  describe('first section', () => {
    it('should rendered correct elements count', () => {
      const actual: number = fixture.debugElement.queryAll(By.css('.example.example--first .item')).length;
      const expected = 100;

      expect(actual).toBe(expected);
    });

    it('should be inactive item', () => {
      const el: DebugElement = fixture.debugElement.query(By.css('.example.example--first .item'));

      component.handleAction({ target: el.nativeElement, visible: false } as InViewportAction);
      fixture.detectChanges();

      const actual = !!el.classes[activeClassName];
      const expected = false;

      expect(actual).toBe(expected);
    });

    it('should be active item', () => {
      const el: DebugElement = fixture.debugElement.query(By.css('.example.example--first .item'));

      component.handleAction({ target: el.nativeElement, visible: true } as InViewportAction);
      fixture.detectChanges();

      const actual: boolean = el.classes[activeClassName];
      const expected = true;

      expect(actual).toBe(expected);
    });
  });

  describe('second section', () => {
    it('should rendered correct elements count', () => {
      const actual: number = fixture.debugElement.queryAll(By.css('.example.example--second .item')).length;
      const expected = 100;

      expect(actual).toBe(expected);
    });

    it('should be inactive item', () => {
      const el: DebugElement = fixture.debugElement.query(By.css('.example.example--second .item'));

      component.handleAction({ target: el.nativeElement, visible: false } as InViewportAction);
      fixture.detectChanges();

      const actual = !!el.classes[activeClassName];
      const expected = false;

      expect(actual).toBe(expected);
    });

    it('should be active item', () => {
      const el: DebugElement = fixture.debugElement.query(By.css('.example.example--second .item'));

      component.handleAction({ target: el.nativeElement, visible: true } as InViewportAction);
      fixture.detectChanges();

      const actual: boolean = el.classes[activeClassName];
      const expected = true;

      expect(actual).toBe(expected);
    });
  });
});
