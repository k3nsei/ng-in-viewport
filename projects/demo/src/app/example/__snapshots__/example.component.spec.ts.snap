// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent should create component 1`] = `
<div
  id="root0"
>
  <section
    class="example example--first"
  >
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       1 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       2 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       3 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       4 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       5 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       6 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       7 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       8 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       9 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       10 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       11 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       12 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       13 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       14 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       15 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       16 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       17 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       18 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       19 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       20 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       21 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       22 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       23 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       24 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       25 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       26 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       27 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       28 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       29 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       30 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       31 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       32 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       33 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       34 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       35 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       36 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       37 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       38 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       39 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       40 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       41 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       42 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       43 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       44 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       45 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       46 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       47 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       48 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       49 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       50 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       51 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       52 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       53 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       54 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       55 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       56 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       57 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       58 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       59 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       60 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       61 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       62 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       63 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       64 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       65 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       66 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       67 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       68 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       69 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       70 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       71 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       72 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       73 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       74 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       75 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       76 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       77 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       78 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       79 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       80 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       81 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       82 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       83 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       84 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       85 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       86 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       87 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       88 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       89 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       90 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       91 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       92 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       93 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       94 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       95 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       96 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       97 
    </div>
    <div
      class="first-section-child item"
      inviewport=""
    >
       98 
    </div>
    <div
      class="first-section-child item item--alt"
      inviewport=""
    >
       99 
    </div>
    <div
      class="first-section-child item item--last"
      inviewport=""
    >
       100 
    </div>
    
  </section>
  <section
    class="example example--second"
  >
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       1 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       2 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       3 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       4 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       5 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       6 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       7 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       8 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       9 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       10 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       11 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       12 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       13 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       14 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       15 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       16 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       17 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       18 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       19 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       20 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       21 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       22 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       23 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       24 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       25 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       26 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       27 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       28 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       29 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       30 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       31 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       32 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       33 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       34 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       35 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       36 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       37 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       38 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       39 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       40 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       41 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       42 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       43 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       44 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       45 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       46 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       47 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       48 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       49 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       50 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       51 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       52 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       53 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       54 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       55 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       56 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       57 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       58 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       59 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       60 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       61 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       62 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       63 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       64 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       65 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       66 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       67 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       68 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       69 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       70 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       71 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       72 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       73 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       74 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       75 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       76 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       77 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       78 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       79 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       80 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       81 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       82 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       83 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       84 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       85 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       86 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       87 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       88 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       89 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       90 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       91 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       92 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       93 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       94 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       95 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       96 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       97 
    </div>
    <div
      class="second-section-child item"
      inviewport=""
    >
       98 
    </div>
    <div
      class="second-section-child item item--alt"
      inviewport=""
    >
       99 
    </div>
    <div
      class="second-section-child item item--last"
      inviewport=""
    >
       100 
    </div>
    
  </section>
</div>
`;
